targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp

```diff
@@ -106,15 +117,49 @@ namespace PlayFab
         return (blockSize * blockCount);
     }
 
-    void PlayFabHttp::AddRequest(const std::string& urlPath, const std::string& authKey, const std::string& authValue, const Json::Value& requestBody, RequestCompleteCallback internalCallback, SharedVoidPointer successCallback, ErrorCallback errorCallback, void* customData)
+    void PlayFabHttp::AddPostRequest(const std::string& url, std::unordered_map<std::string, std::string> headers, const std::string& requestBody, RequestCompleteCallback internalCallback, ErrorCallback userErrorCallback, void* extraBuffer)
+    {
+        std::string authValue = "";
+        std::string authKey = "";
+        for (auto const &obj : headers)
+        {
+            if (obj.first == "X-SecretKey" || obj.first == "X-Authorization" || obj.first == "X-EntityToken")
+            {
+                authKey = obj.first;
+                authValue = obj.second;
+                break;
+            }
+        }
+
+        Json::Value root;
+        Json::Reader reader;
+        bool parsingSuccessful = reader.parse(requestBody.c_str(), root);
+
+        if (parsingSuccessful)
+        {
+            InternalAddRequest(url, authKey, authValue, root, internalCallback, userErrorCallback, extraBuffer);
+        }
+        else
+        {
+            throw new std::invalid_argument("The requestBody string was not in a valid Json format during AddPostRequest.");
+        }
+    }
+
+    void PlayFabHttp::InternalAddRequest(const std::string& urlPath, const std::string& authKey, const std::string& authValue, const Json::Value& requestBody, RequestCompleteCallback internalCallback, ErrorCallback errorCallback, void* customData)
     {
         CallRequestContainer* reqContainer = new CallRequestContainer();
         reqContainer->errorWrapper.UrlPath = urlPath;
         reqContainer->authKey = authKey;
         reqContainer->authValue = authValue;
         reqContainer->errorWrapper.Request = requestBody;
         reqContainer->internalCallback = internalCallback;
-        reqContainer->successCallback = successCallback;
+
+        if (sCurrentCallback != nullptr)
+        {
+            reqContainer->successCallback = sCurrentCallback;
```

Comment:
If we need to track additional callbacks as part of the InternalAddRequest signature, that's fine. We should NOT be providing this parameter via a static parameter. That's just going to cause concurrency or threading issues.
The nice thing about InternalAddRequest is that we can change the signature however we need, so just add it as a parameter. Don't do this roundabout static field as a property thing.